src/main/java/at/ac/tuwien/kr/alpha/solver/heuristics/HeapOfActiveAtoms.java

```diff
@@ -194,8 +214,12 @@ private void initActivityNaive(NoGood newNoGood) {
 	/**
 	 * Returns the atom with the highest activity score and removes it from the heap.
 	 */
-	public Integer getMostActiveAtom() {
-		return heap.poll();
+	Integer getMostActiveAtom() {
+		Integer mostActiveAtom = heap.poll();
+		if (mostActiveAtom != null) {
+			occursInHeap[mostActiveAtom] = false;
+		}
+		return mostActiveAtom;
```

Comment:
Should we search the heap for the first atom for which `occursInHeap[...] == true`? (see the comment in line 255)

Reply:
Yes, done as per our discussion.